File: app/java/src/main/java/com/example/production/domain/model/inventory/StockStatus.java
package com.example.production.domain.model.inventory;

import lombok.Getter;
import lombok.RequiredArgsConstructor;

/**
 * 在庫状態
 */
@Getter
@RequiredArgsConstructor
public enum StockStatus {
    PASSED("合格"),
    DEFECTIVE("不良"),
    UNINSPECTED("未検査");

    private final String displayName;

    public static StockStatus fromDisplayName(String displayName) {
        for (StockStatus status : values()) {
            if (status.displayName.equals(displayName)) {
                return status;
            }
        }
        throw new IllegalArgumentException("Unknown stock status: " + displayName);
    }
}

File: app/csharp/src/ProductionManagement.Domain/Models/Inventory/StockStatus.cs
namespace ProductionManagement.Domain.Models.Inventory;

/// <summary>
/// 在庫状態
/// </summary>
public enum StockStatus
{
    Passed,
    Defective,
    Uninspected
}

public static class StockStatusExtensions
{
    private static readonly Dictionary<StockStatus, string> DisplayNames = new()
    {
        { StockStatus.Passed, "合格" },
        { StockStatus.Defective, "不良" },
        { StockStatus.Uninspected, "未検査" }
    };

    public static string ToDisplayName(this StockStatus status)
        => DisplayNames[status];

    public static StockStatus FromDisplayName(string displayName)
        => DisplayNames.First(x => x.Value == displayName).Key;
}
